src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,262 @@
+        """Initialize the RPNG object from a 4-character or 5-character string
+        
+        4-character -> simplified format -> RPNG
+        5-character -> extended format -> RAPNG
+
+        Note that the ancilla basis(a) is the second character, just before (p).
+
+        Note that any character different from a BasisEnum / ExtendedBasisEnum
+        value would result in the corresponding field being None.
+        """
+        a_str = 'z'
+        if len(rpng_string) == 4:
+            r_str, p_str, n_str, g_str = tuple(rpng_string)
+        elif len(rpng_string) == 5:
+            r_str, a_str, p_str, n_str, g_str = tuple(rpng_string)
+        else:
+            raise ValueError('The rpng string must be exactly 4- or 5-character long.')
+        # Convert the characters into the enum attributes (or raise error).
+        r = ExtendedBasisEnum(r_str) if r_str in ExtendedBasisEnum._value2member_map_ else None
+        a = BasisEnum(a_str) if a_str in BasisEnum._value2member_map_ else None
+        p = BasisEnum(p_str) if p_str in BasisEnum._value2member_map_ else None
+        n = int(n_str) if n_str.isdigit() else None
+        g = ExtendedBasisEnum(g_str) if g_str in ExtendedBasisEnum._value2member_map_ else None
+        # Raise error if anythiong but '-' was used to indicate None.
+        if not r and r_str != '-':
+            raise ValueError('Unacceptable character for the R field.')
+        if not p and p_str != '-':
+            raise ValueError('Unacceptable character for the P field.')
+        if p and not a:
+            raise ValueError('Unacceptable ancilla basis in the extended format (i.e. rapng).')
+        if not n and n_str != '-':
+            raise ValueError('Unacceptable character for the N field.')
+        if not g and g_str != '-':
+            raise ValueError('Unacceptable character for the G field.')
+        return cls(r, p, n, g, a)
+
+    
+    def get_r_op(self) -> str | None:
+        """Get the reset operation or Hadamard"""
+        op = self.r
+        if op == None:
+            return None
+        elif op.value in BasisEnum._value2member_map_:
+            return f'R{op.value.upper()}'
+        else:
+            return f'{op.value.upper()}'
+
+    def get_g_op(self) -> str | None:
+        """Get the measurement operation or Hadamard"""
+        op = self.g
+        if op == None:
+            return None
+        elif op.value in BasisEnum._value2member_map_:
+            return f'M{op.value.upper()}'
+        else:
+            return f'{op.value.upper()}'
+
+
+@dataclass
+class RGN:
+    """Organize the prep and meas bases for the ancilla, together with the meas time
+    
+    The init time is assumed to be 0.
+
+    Attributes:
+        r   ancillaqubit reset basis  
+        g   ancilla qubit measure basis
+        n   time step of ancilla measurement
+    """
+    r: BasisEnum = BasisEnum.X
+    g: BasisEnum = BasisEnum.X
+    n: int = 6
```

Comment:
I am pretty sure `n` should not be included in this representation.
```suggestion
```
The problem of including `n` is that it forces the user to have a value (either user-provided, or the default `6`), which is highly misleading because we will eventually impose that resets and measurements all happen at the same timestep at compilation, so the specific value provided here will likely not be respected.

Reply:
We need to know it when we create the circuit associated with the plaquette, so it should be available then.
My idea is that it can be determined before creating the plaquette.
Alternatively we can add it as argument to the method generating the plaquette (and its circuit), adding checks of compatibility with the n in the corners.

Reply:
I am fine having that information somewhere. I agree that we will eventually need it. My main issue is that this information should not be accessible and "settable" by the user, because `tqec` internals will set the correct value for it, and so giving the user the impression he/she can fix the measurement time-step would be counter-intuitive.